src/option.c

```diff
@@ -4081,11 +4081,9 @@ did_set_showtabline(optset_T *args UNUSED)
     char *
 did_set_smoothscroll(optset_T *args UNUSED)
 {
-    if (curwin->w_p_sms)
-	return NULL;
+    if (!curwin->w_p_sms)
+	curwin->w_skipcol = 0;
 
-    curwin->w_skipcol = 0;
-    changed_line_abv_curs();
```

Comment:
> can we have a test for this?

Added one. Also removed this `changed_line_abv_curs()` call for `did_set_smoothscroll()` because it is a `P_RWIN` option and it is already called in `changed_window_setting()`.